datapower-docker-compose-dp-mq/datapower-docker-compose-dp-mq.mkd

```diff
@@ -0,0 +1,134 @@
+
+
+## The Composed Application
+No rock band here, just the message Bob. Harley's Hello World example (mentioned in the Prerequisite section) goes into great detail about the various elements of his project. In order to avoid redundancy, I will not go into such detail here, but I will still cover the whole project. Let's take a look at the `docker-compose.yml` file that orchestrates the entire environment.
+```
+version: '2'
+services:
+    mq:
+      build: mq
+    http:
+      build: http
+    datapower:
+      build: datapower
+      depends_on:
+       - mq
+       - http
+    soapui:
+      image: ddavison/soapui:latest
+      depends_on:
+       - datapower
+    curldriver:
+      build: curldriver
+      depends_on:
+       - soapui
+```
+
+In all, the environment consists of five services, which build five images, and run five containers. 
+The `curldriver` service builds an image with curl installed from an Ubuntu image.
+The `soapui` service builds a SoapUI image from Docker Hub.
+The `datapower` service builds a fully configured DataPower, with an application domain that contains a multi-protocol gateway, from ibmcom/datapower.
+The `http` service builds an apache server image from Docker Hub. 
+The `mq` service builds a configured MQ queue manager, built off a base MQ image, from Docker Hub.
+
+The general flow is a message is sent via curl to a SoapUI container which sends a test request to DataPower. The request is transformed by a processing policy in DataPower, which retrieves and utilizes style sheets and schema from a web server. Finally, the resulting message is sent from DataPower to a back end MQ queue manager. An assertion on the response is evaluated by SoapUI and a message is returned with the outcome of the test.
+
+From the docker-compose.yml file, you can see that the services have dependencies on one another. As in the Hello World docker compose example, the networking in the self-contained environment is configured and controlled by Docker. All of the containers are aware of each other in the network and the different nodes can be referred by one another as they are named by the `docker-compose.yml`.
+
+## Run the Application
+In order to run the application, you need to get the source. Run the following commands:
+```
+git clone https://github.com/ibm-datapower/datapower-tutorials.git
+cd datapower-tutorials/datapower-docker-compose-dp-mq/src
+```
+
+We'll name the project bob and build it with the following command:
+```
+docker-compose -p bob build
+```
+
+This builds four images as you can see below:
+```
+$ docker images
```

Comment:
If you named your project bob, then the images will have the project name pre-pended, i.e
NOT `foo_datapower` but rather `bob_datapower` in this case